test-case/multiple-pipeline.sh

```diff
@@ -105,7 +105,7 @@ func_run_pipeline_with_type()
 
         dlogi "Testing: $pcm [$dev]"
 
-        "${APP_LST[$1]}" -D "$dev" -c "$channel" -r "$rate" -f "$fmt" "${DEV_LST[$1]}" -q &
+        "${APP_LST[${1%% *}]}" -D "$dev" -c "$channel" -r "$rate" -f "$fmt" "${DEV_LST[${1%% *}]}" -q &
```

Comment:
Good idea but this is really too cryptic, it took me several minutes to understand what this does. Please use more English like this instead:

```shell
# args: $1: playback or capture
#          $2: optional filter

func_run_pipeline_with_type()
{
     local direction="$1"
     local opt_filter
     if [ -n "$2" ]; then opt_filter="& $2"; fi
     ...
     func_pipeline_export "$tplg" "type:$direction $opt_filter"
     ...
    "${APP_LST[$direction]}" -D "$dev" -c "$channel" -r "$rate" -f "$fmt" "${DEV_LST[$direction]}" -q &
     ...
}


func_run_pipeline_with_type 'capture'  '~pcm:Amplifier Reference'

```

Reply:
Thanks Marc. it's simple and easy to understand. I have pushed a new version.